### ISSUE TITLE
Can't reproduce results of KEGG dataset

### ISSUE THREAD

**Issue by JadenNeal** [bowang-lab/BioReason#16]
Hi, thank you for your great work!
I'm trying to reproduce the results of BioReason on the KEGG dataset of train_dna_only task.
However, when I use the ntv2-50m and ntv2-500m models, the test_f1, test_recall and test_precision are always 0, even though I used the same parameters as reported in the paper.
My trial:

1. I installed your pip-env without errors.
2. add `sorted` [here](https://github.com/bowang-lab/BioReason/blob/main/train_dna_only.py#L285) so that I can get the determinstic results.
3. I tried ntv2-500m and ntv2-50m, different learning rate(default=3e-4 in you codebase, but 5e-5 in your paper, I tried both), and different batch_size. I even turned the `train_just_classifier` into `Fasle` and remove `sorted` fucntion described in trial 2. But the test results still be 0(except test_acc)

Below are results of ntv2-50m on KEGG:
━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│      test_acc_epoch       │    0.8275862336158752     │
│       test_f1_epoch       │            0.0            │
│      test_loss_epoch      │    0.6577547192573547     │
│   test_precision_epoch    │            0.0            │
│     test_recall_epoch     │            0.0            │
└───────────────────────────┴───────────────────────────

Could you please let me know if there are any special settings required for the KEGG dataset, or if this issue has been observed before?

Thanks in advance for your help!

**Comment by DBpackage** [comment]
Dear JadenNeal,

I am writing to ask for tips on successfully setting up a Conda environment that includes both trl and vllm.

Currently, I am struggling with installation due to version conflicts between these two libraries. If anyone has a working environment configuration or a specific set of compatible versions (requirements.txt or environment.yml), could you please share your insights?

Thanks in advance!

**Comment by JadenNeal** [comment]
> Dear JadenNeal,
> 
> I am writing to ask for tips on successfully setting up a Conda environment that includes both trl and vllm.
> 
> Currently, I am struggling with installation due to version conflicts between these two libraries. If anyone has a working environment configuration or a specific set of compatible versions (requirements.txt or environment.yml), could you please share your insights?
> 
> Thanks in advance!

Hi DBpackage,
I did not install trl and vllm. I just tried the SFT experiments and found it impossible to **reproduce**, so I did not continue with the subsequent experiments.
